contrib/camel/pkg/apis/sources/v1alpha1/camelsource_types.go

```diff
@@ -0,0 +1,165 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/knative/pkg/apis/duck"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CamelSource is the Schema for the camelsources API
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:categories=all,knative,eventing,sources
+type CamelSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CamelSourceSpec   `json:"spec,omitempty"`
+	Status CamelSourceStatus `json:"status,omitempty"`
+}
+
+// Check that CamelSource can be validated and can be defaulted.
+var _ runtime.Object = (*CamelSource)(nil)
+
+// Check that CamelSource implements the Conditions duck type.
+var _ = duck.VerifyType(&CamelSource{}, &duckv1alpha1.Conditions{})
+
+const (
+	// CamelSourceConditionReady has status True when the CamelSource is ready to send events.
+	CamelConditionReady = duckv1alpha1.ConditionReady
+
+	// CamelConditionSinkProvided has status True when the CamelSource has been configured with a sink target.
+	CamelConditionSinkProvided duckv1alpha1.ConditionType = "SinkProvided"
+
+	// CamelConditionDeployed has status True when the CamelSource has had it's deployment created.
+	CamelConditionDeployed duckv1alpha1.ConditionType = "Deployed"
+)
+
+var camelCondSet = duckv1alpha1.NewLivingConditionSet(
+	CamelConditionSinkProvided,
+	CamelConditionDeployed,
+)
+
+// CamelSourceSpec defines the desired state of CamelSource
+type CamelSourceSpec struct {
+	// Source is the reference to the integration flow to run.
+	Source CamelSourceOriginSpec `json:"source"`
+
+	// ServiceAccountName is the name of the ServiceAccount to use to run this
+	// source.
+	// +optional
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
+
+	// Image is an optional base image used to run the source.
+	// +optional
+	Image string `json:"image,omitempty"`
+
+	// Sink is a reference to an object that will resolve to a domain name to use as the sink.
+	// +optional
+	Sink *corev1.ObjectReference `json:"sink,omitempty"`
+}
+
+type CamelSourceOriginSpec struct {
+	Component *CamelSourceOriginComponentSpec `json:"component,omitempty"`
+}
+
+type CamelSourceOriginComponentSpec struct {
+	// URI is a Camel component URI to use as starting point
+	// +kubebuilder:validation:MinLength=1
+	URI        string            `json:"uri,omitempty"`
+	Properties map[string]string `json:"properties,omitempty"`
```

Comment:
Camel supports query parameters and global properties. Query parameters are written directly in the URI, while properties are usually global Camel options or component options. Both are documented in the Camel github repo doc.

Reply:
Should this reference the camel-k github repo more explicitly in the readme?

In general, I'd rather see the `README.md` describe how to install standard `camel-k` and then these components, rather than copying (vendoring) some `camel-k` components with changes (including with a `kubectl apply -f https://path-to-camel-k/release.yaml https://knative-eventing-nightly/releases/...`). Using vendored components with local changes makes it difficult for users to figure out whether they can "upgrade" to a full `camel-k` with other components from their Knative setup, or whether the Knative modifications are material and conflict with upstream.

Reply:
Agree, will move the camel k installation out and keep here only the api package to communicate